text/3503-frontmatter.md

```diff
@@ -0,0 +1,579 @@
+use actix_web::Responder;
+use actix_web::web;
+use tokio::spawn;
+use tokio::sync::oneshot;
+use tokio::task::spawn_blocking;
+```
+
+## PL/Rust
+
+Example:
+```sql
+CREATE OR REPLACE FUNCTION randint() RETURNS bigint LANGUAGE plrust AS $$
+[dependencies]
+rand = "0.8"
+
+[code]
+use rand::Rng; 
+Ok(Some(rand::thread_rng().gen())) 
+$$;
+```
+
+See [External Dependencies](https://github.com/tcdi/plrust/blob/main/doc/src/dependencies.md)
+
+## YAML frontmatter
+
+As a specialization of [YAML presentation streams](https://yaml.org/spec/1.2.2/#323-presentation-stream),
+static site generators use frontmatter to embed YAML at the top of files.
+Other systems have extended this for non-YAML use, like
+[zola using `+++` for TOML](https://www.getzola.org/documentation/content/page/#front-matter).
+
+## Proposed Python syntax
+
+Currently the draft [PEP 723](https://peps.python.org/pep-0723/) proposes allowing begin/end markers inside of regular comments:
+
+```python
+# /// pyproject
+# [run]
+# requires-python = ">=3.11"
+# dependencies = [
+#   "requests<3",
+#   "rich",
+# ]
+# ///
+
+import requests
+from rich.pretty import pprint
+
+resp = requests.get("https://peps.python.org/api/peps.json")
+data = resp.json()
+pprint([(k, v["title"]) for k, v in data.items()][:10])
+```
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+- Treat `cargo` as the default infostring
+  - Since this is a one-way door, we want to make sure we are confident that that is the right default / policy
+  - If there aren't guard rails (e.g. rustc enforcement) that no-infostring
+    means `cargo`, it can easily be a free-for-all and we need to decide if we
+    want that or if there is enough guard rails to discourage it.
+- Support more infostring languages
+  - We need to better understand use cases for how this should be extended, particularly what the syntax should be
+    - For reference, infostrings in commonmark (language + attributes) is just
+      all non-backtick characters, supports `\` escapes, and entity / numeric
+      character references with a vague notion of the language being the first "word"
+    - A safe starting point could be to say that it must be an identifier
+  - Hard coding it for now is much like how rustc has hard coded support for `#[rustdoc]` or `clippy`/`rustdoc` lint categories
+- Support infostring attributes
+  - We need to better understand use cases for how this should be extended, particularly what the syntax should be (see infostring language)
+  - A safe starting point could be to say that a space or comma separates attributes and everything after it is defined as part of the "language"
+- Loosen the code-fence syntax, like allowing newlines
+- Add support for a `#[frontmatter(info = "", content = "")]` attribute that this syntax maps to.
+  - Since nothing will read this, whether we do it now or in the future will have no affect
```

Comment:
I think we should specify this in the RFC, so it's clear what it should map to in Rust ASTs (just like doc comments map to [`syn::Attribute`](https://docs.rs/syn/latest/syn/struct.Attribute.html)).